### PR TITLE
MAINT: Cleanup assortativity module, remove unused variables

### PR DESCRIPTION
**Pull request by MridulS**
Raise errors instead of accepting invalid arguments silently

This would also help with less errors for folks using linters locally in their IDEs

### CODE REVIEW

File: networkx/algorithms/assortativity/correlation.py [hunk 1]
```diff
@@ -150,12 +149,11 @@ def degree_pearson_correlation_coefficient(G, x="out", y="in", weight=None, node
     .. [2] Foster, J.G., Foster, D.V., Grassberger, P. & Paczuski, M.
        Edge direction and the structure of networks, PNAS 107, 10815-20 (2010).
     """
-    import scipy as sp
-    import scipy.stats  # call as sp.stats
+    import scipy.stats
```
Comment (jarrodmillman) [inline networkx/algorithms/assortativity/correlation.py:152]: Let's keep using this import pattern.  It was put in place to prepare for upcoming changes using lazy loading.

File: networkx/algorithms/assortativity/pairs.py [hunk 2]
```diff
@@ -48,10 +48,10 @@ def node_attribute_xy(G, attribute, nodes=None):
         if G.is_multigraph():
             for v, keys in nbrsdict.items():
                 vattr = Gnodes[v].get(attribute, None)
-                for k, d in keys.items():
+                for _ in range(len(keys)):
```
Comment (dschult) [inline networkx/algorithms/assortativity/pairs.py:51]:
```suggestion
                for _ in keys:
```
Was the `range(len(` part added for a reason? Maybe readability? Is this suggestions better?

**Review by dschult — APPROVED** [review]
This looks good. One question below. I'm fine with leaving as is or switching.

File: networkx/algorithms/assortativity/pairs.py (hunk 2 above)
Comment (MridulS) [reply networkx/algorithms/assortativity/pairs.py]: this is much better :)

File: networkx/algorithms/assortativity/connectivity.py [hunk 3]
```diff
@@ -72,9 +75,9 @@ def average_degree_connectivity(
     # First, determine the type of neighbors and the type of degree to use.
     if G.is_directed():
         if source not in ("in", "out", "in+out"):
-            raise ValueError('source must be one of "in", "out", or "in+out"')
+            raise nx.NetworkXError('source must be one of "in", "out", or "in+out"')
```
Comment (rossbar) [inline networkx/algorithms/assortativity/connectivity.py:78]: Technically changing the exception type here might bite users who are catching `ValueError`s - is there a strong motivation for doing so?

Comment (rossbar) [reply networkx/algorithms/assortativity/connectivity.py:78]: Ah I see - it looks like the goal is to match the exception types in `neighbor_degree`. This seems fine to me then: +1 for consistency and one of the exception types will have to change so might as well be this one!

**Review by rossbar — APPROVED** [review]
LGTM, thanks @MridulS . If we wanted to be *extra* careful you could add a quick bullet to the `API Changes` in the release notes about the change in exception type for `average_degree_connectivity`.

**Pull request merged**